app/models/host/managed.rb

```diff
@@ -658,6 +658,12 @@ def hash_clone(value)
   def set_hostgroup_defaults
     return unless hostgroup
     assign_hostgroup_attributes(%w{domain_id compute_profile_id})
+
+    if compute_attributes && compute_profile_id != compute_profile_id_was
```

Comment:
[Extract method](http://refactoring.com/catalog/extractMethod.html) :smile: 
Why does it clear all `compute_attributes` before calling `set_compute_attributes` though?